### ISSUE TITLE
Empty output on development environment

### ISSUE THREAD

**Issue by almod90** [laravel-admin-extensions/composer-viewer#2]
Package installed successfully, however it returns empty output.

Probable reason for this is **xdebug** enabled in development environment [as refered here](https://stackoverflow.com/questions/45815117/why-cant-i-run-the-composer-script-from-php-using-exec-shell-exec-system)

Checked your composer execution code under **artisan tinker** and it works fine, so it's xdebug issue 99%

I would appreciate if you'd be able to solve this issue

**Comment by yurik94** [comment]
+1

**Comment by yurik94** [comment]
![image](https://user-images.githubusercontent.com/13189646/47607972-779a9800-da27-11e8-9b2a-dc0b8c722f6b.png)

After some debug it seems that I don't have composer in path, check that!

**Comment by jxlwqq** [comment]
https://getcomposer.org/doc/articles/troubleshooting.md#xdebug-impact-on-composer
Is helpfull?

**Comment by AaronChengHao** [comment]
![image](https://user-images.githubusercontent.com/20119217/51439774-233b4780-1cfa-11e9-8e64-b68efb529d4f.png)
composer 环境变量不能识别，我加了一个全路径就可以了

**Comment by AaronChengHao** [comment]
![image](https://user-images.githubusercontent.com/20119217/51439917-c04ab000-1cfb-11e9-9140-51b61059168d.png)
![image](https://user-images.githubusercontent.com/20119217/51439924-c8a2eb00-1cfb-11e9-876e-2c8a44e56455.png)

我不知道为什么没有执行成功

**Comment by schel4ok** [comment]
also doesn't work for me

**Comment by jxlwqq** [comment]
https://github.com/laravel-admin-extensions/composer-viewer/commit/1b6790d91fb89dd489bf3908dfe4ef8972f0808a

hope this commit can be helpful.

**Comment by schel4ok** [comment]
no. I still have empty Composer Packages list on `/admin/composer-viewer` page

**Comment by schel4ok** [comment]
any ideas?

**Comment by schel4ok** [comment]
@jxlwqq please help

**Comment by jxlwqq** [comment]
PHP.INI settings Disable exec, shell_exec, system, popen and Other Functions To Improve Security.

Check your php.ini file, and make sure `exec` function is removed.

![Lark20200605193255](https://user-images.githubusercontent.com/2421068/83871848-65d0b900-a763-11ea-8578-09cfe66f9983.png)

**Comment by schel4ok** [comment]
I have empty value
```
disable_functions = 
```

**Comment by usamaits** [comment]
First of all you have to enable PHP exec() in your php.ini  if it's not ,

it's work fine, typically **'which-composer'** asks for **full composer file path** + **file name** ex;
'c:\dev\php composer\composer'.

I'm using Windows.
So, if **composer** is added on global system path ; 'composer file run globally from any path', 
just replace config/admin variable **'which-composer'** by **'composer'.**

 'composer-viewer' => [
            // Set this to false if you want to disable this extension
            'enable' => true,
            // Set the location of composer command
            'which-composer' => 'composer', // !! it's important !!
        ]

you have to fulfill and pass this two line to run composer : 
**$command = 'cd ' . base_path() . ' && ' . $which_composer . ' show --latest --format=json';
exec($command, $output);**
